docs/guide/ethereum-provider.md

```diff
@@ -1,103 +1,164 @@
+[read more about here](https://medium.com/metamask/breaking-changes-to-the-metamask-inpage-provider-b4dde069dd0a).
 
-### ethereum.isMetaMask
+At that time, we will:
 
-Returns `true` or `false`, representing whether the user has MetaMask installed.
+- Stop emitting `chainIdChanged`, and instead emit `chainChanged`
+- Ensure that all `chainId` values are **not** 0-prefixed
+  - For example, instead of `0x01`, we will always return `0x1`, wherever the chain ID is returned or accessible.
+- Remove the following experimental methods:
+  - `ethereum._metamask.isEnabled`
+  - `ethereum._metamask.isApproved`
 
-### ethereum.autoRefreshOnNetworkChange
+These changes _may_ break your website.
+Please read our [migration guide](about:blank) for more details.
 
-When the network is changed, MetaMask will reload any pages that have made requests to the provider.
-This automatic reload behavior will be removed in a future release of MetaMask, but in the meantime it can be disabled with this flag.
+### `window.web3` Removal
 
-To disable auto-refresh on a network change you can do:
+::: tip
+If you do not use the `window.web3` object injected by MetaMask, you will not be affected by these changes.
+:::
 
-```javascript
-ethereum.autoRefreshOnNetworkChange = false;
-```
+In **Q4 2020** (date TBD), we will:
 
-This can be toggled on or off at any time.
+- Stop injecting `window.web3` into web pages
+- Remove the `ethereum.autoRefreshOnNetworkChange` property
 
-### ethereum.networkVersion (DEPRECATED)
+If you rely on the `window.web3` object currently injected by MetaMask, these changes _will_ break your website.
+Please read our [migration guide](about:blank) for more details.
+
+## Basic Usage
+
+For any non-trivial Ethereum web application — a.k.a. web3 site — to work, you will have to:
+
+- Detect the Ethereum provider (`window.ethereum`)
+- Detect which Ethereum network the user is connected to
+- Get the user's Ethereum account(s)
+
+The snippet at the top of this page is sufficient for detecting the provider.
+You can learn how to accomplish the other two by reviewing the snippet in the [Using the Provider section](#using-the-provider).
+
+Although any Ethereum operation can be performed via the provider API, most developers use a convenience library with higher-level abstractions.
+The most common such libraries are [ethers](https://www.npmjs.com/package/ethers) and [web3.js](https://www.npmjs.com/package/web3).
+
+Even if you use a convenience library, you will likely still need to detect the provider before initializing the library.
+Other than that, you can generally learn everything you need to know from the documentation of those libraries, without reading this lower-level API.
+
+However, for developers of convenience libraries, and for developers who would like to use features that are not yet supported by their favorite libraries, knowledge of the provider API is essential. Read on for more details.
```

Comment:
I'm not a fan of how this is framed 🤔 This seems like we're discouraging most devs from using the provider directly, and I don't think we should do that. We should reference convenience libraries, but we don't need to describe them as "high-level" and us as "low-level", and we don't need to make broad generalizations about who might want to use us directly.

Any dev who trusts the EIP-1193 standard to remain stable more than they trust `web3` or `ethers` to remain stable might want to use the provider directly. Or anyone that wants to just use fewer dependencies.

Reply:
I tried and failed to find a way to re-word this to avoid those problems. Maybe this whole section could be omitted from this page. We already mention convenience libraries in the 'Getting Started' guide.

Reply:
How about this?
------------------------------------
> The provider API is all you need to create a full-featured web3 application.
> 
> That said, many developers use a convenience library, such as [ethers](https://www.npmjs.com/package/ethers) and [web3.js](https://www.npmjs.com/package/web3), instead of using the provider directly.
> If you are in need of higher-level abstractions than those provided by this API, we recommend that you seek out a convenience library.

Reply:
That seems fine! We can always improve it later if we think of a better way to frame it.

Reply:
482ebb5